docs/release-notes/charmcraft-3.5.rst

```diff
@@ -0,0 +1,161 @@
+Charmcraft 3.5 release notes
+============================
+
+**UNRELEASED**. Expected release date: 7 April 2025
```

Comment:
As-is, this will build on `latest`. I don't think we want that.

It would be easier to de-index the page and add it to the blocklist.